app/models/contact.rb

```diff
@@ -12,4 +12,28 @@ class Contact < ActiveRecord::Base
   validates :zip, presence: true
   validates :zip, length: { is: 5 }
   validates :phone, length: { is: 10 }
+
+  def geo_address
+    [self.address, self.city, self.state, self.zip].reject(&:empty?).join(" ")
+  end
+
+  def display_address
+    a = [self.address, self.city].join(" | ")
+    b = [a, self.state].join(", ")
+    [b, self.zip].join(" | ")
+  end
+
+  def display_phone
+    phone = self.phone.to_s.split('')
+    [phone[0..2].join(''),
+    phone[3..5].join(''),
+    phone[6..9].join('')].join(".") + " | ext " + self.phone_ext
```

Comment:
Align the elements of an array literal if they span more than one line.<br>Redundant `self` detected.